### PR TITLE
EVA-1937 — Tests for the repeat expansion pipeline

### PR DESCRIPTION
**Pull request by tskir**
This PR is based on top of #3 and includes all changes from it as well. To view only changes from this PR, see this [**diff link**](https://github.com/tskir/vep-mapping-pipeline/compare/repeat-expansion-variants...eva-1937-repeat-expansion-tests?expand=1).

* Implemented the tests
  + Added a test dataset which includes all NT expansion variants from ClinVar. Records of special interest are moved to the top and annotated. See more details in `repeat_expansion_variants/tests/README.md`
  + Implemented the integration test using `pytest`

* Small adjustments to the pipeline
  + Fixed a regexp
  + Modified column order

* Technical changes
  + Updated configuration files: `.gitignore`, `.idea`
  + Simplified installation instructions for Travis
  + Added `setup.py` and `__init__.py` to make the repository a proper package (required for running the tests)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+# The pipeline requires GNU parallel; bcftools; and certain Python modules
 install:
-  - pip -q install retry requests
-  - git clone -q git://github.com/samtools/htslib.git
-  - git clone -q git://github.com/samtools/bcftools.git
-  - cd bcftools && make --quiet --jobs `nproc` && cd ..
-  - export PATH=$PATH:bcftools
+  - sudo apt update
+  - sudo apt -y install samtools bcftools parallel libbz2-dev liblzma-dev
+  - pip -q install -r requirements.txt
```
Comment (sundarvenkata-EBI) [inline .travis.yml:10]: Why the "-q" switch? We want the output to be verbose for troubleshooting, right?

File: README.md [hunk 2]
```diff
@@ -1,74 +1,20 @@
...
+## Installing requirements
 
-### Input file
-The pipeline uses succinct, VCF-compatible variant identifiers, in the format of `CHROM:POS:REF:ALT`. Input file must contain variants in exactly this format, one entry per line. Example:
-```
-10:110821920:G:A
-10:110821933:C:T
-10:110821934:G:A
-```
+The commands below has been tested for Ubuntu 19.10. You might have to adjust commands and package names if you're using
```
Comment (sundarvenkata-EBI) [inline README.md]: Wait, the travis YML says "bionic" which is Ubuntu 18.04...??

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+setup(
+    name='vep-mapping-pipeline',
+    version='',
+    packages=['vep_mapping_pipeline', 'repeat_expansion_variants', 'repeat_expansion_variants.test'],
+    url='',
+    license='',
+    author='ktsukanov',
+    author_email='ktsukanov@ebi.ac.uk',
+    description=''
```
Comment (sundarvenkata-EBI) [inline setup.py]: Suggest adding a description here...

File: .travis.yml (hunk 1 above)
Comment (tskir) [reply .travis.yml:10]: `-q` only affects info-level messages, which we don't want. If anything does go wrong, it'll report it (and also throw a non-zero return code), for example:

```
$ pip3 -q install nonexistent_package
ERROR: Could not find a version that satisfies the requirement nonexistent_package (from versions: none)
ERROR: No matching distribution found for nonexistent_package
$ echo $?
1
```

File: README.md (hunk 2 above)
Comment (tskir) [reply README.md]: Haha, fair point. Sorry for the confusion. I did initially test everything on Ubuntu 19.10, which my desktop is running. Then I wrote Travis config and it turned out that the latest version available is bionic, 18.04. Luckily it already includes decent package versions and everything works on it as well.

I have changed wording to "tested for Ubuntu 18.04 and newer". Fixed in https://github.com/EBIvariation/vep-mapping-pipeline/pull/4/commits/02e91604f830591c43d9c13404a938f328aa8f09

File: setup.py (hunk 3 above)
Comment (tskir) [reply setup.py]: Fixed in https://github.com/EBIvariation/vep-mapping-pipeline/pull/4/commits/736feddd6e3df72350f37a735d668fe6f905f412

File: README.md [hunk 4]
```diff
@@ -1,74 +1,20 @@
...
 
-### Input file
-The pipeline uses succinct, VCF-compatible variant identifiers, in the format of `CHROM:POS:REF:ALT`. Input file must contain variants in exactly this format, one entry per line. Example:
-```
-10:110821920:G:A
-10:110821933:C:T
-10:110821934:G:A
-```
+The commands below has been tested for Ubuntu 18.04 and newer. You might have to adjust commands and package names if
+you're using a different distribution. Note in particular that some older Debian and Ubuntu distrubutions include
```
Comment (jmmut) [inline README.md]:
```suggestion
you're using a different distribution. Note in particular that some older Debian and Ubuntu distributions include
```

**Review by sundarvenkata-EBI — APPROVED** [review]

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  variants."""
+
+setup(
+    name='vep-mapping-pipeline',
+    version='',
+    packages=['vep_mapping_pipeline', 'repeat_expansion_variants', 'repeat_expansion_variants.test'],
+    url='',
+    license='',
+    author='ktsukanov',
+    author_email='ktsukanov@ebi.ac.uk',
```
Comment (jmmut) [inline setup.py]: I don't want to take authorship from you, but if anyone uses this pipelines, does it make sense to redirect them to eva-dev? I don't really know how this package setup works.

**Review by jmmut — APPROVED** [review]
I don't claim to understand all the details of the test cases, but it looks alright.

File: setup.py (hunk 5 above)
Comment (tcezard) [reply setup.py]: I would leave the `author` and just change the `author_email`.

**Review by tcezard — APPROVED** [review]
It all looks good

File: README.md (hunk 4 above)
Comment (tskir) [reply README.md]: Fixed in https://github.com/EBIvariation/vep-mapping-pipeline/pull/4/commits/80436465c9097037810d32912069efac49e8268f, thanks!

File: setup.py (hunk 5 above)
Comment (tskir) [reply setup.py]: I didn't indent to claim solo authorship :-) This file was originally automatically generated using my PyCharm config, this is where the values come from.

Per Tim's suggestion, changed the `author` to “Kirill Tsukanov, European Variation Archive, EMBL-EBI” and `author_email` to eva-dev@ebi.ac.uk. Fixed in https://github.com/EBIvariation/vep-mapping-pipeline/pull/4/commits/80436465c9097037810d32912069efac49e8268f.

**Pull request merged**